rush.json

```diff
@@ -1,37 +1,255 @@
+   * occasionally, but if that's painful, it's a warning sign that your development style may
+   * discourage refactoring.  Reorganizing the categories should be an enlightening discussion
+   * that brings people together, and maybe also identifies poor coding practices (e.g. file
+   * references that reach into other project's folders without using NodeJS module resolution).
+   *
+   * The defaults are projectFolderMinDepth=2 and projectFolderMaxDepth=2.
+   *
+   * To remove these restrictions, you could set projectFolderMinDepth=1
+   * and set projectFolderMaxDepth to a large number.
+   */
+  // "projectFolderMinDepth": 2,
+  // "projectFolderMaxDepth": 2,
+
+  /**
+   * This feature helps you to review and approve new packages before they are introduced
+   * to your monorepo.  For example, you may be concerned about licensing, code quality,
+   * performance, or simply accumulating too many libraries with overlapping functionality.
+   * The approvals are tracked in two config files "browser-approved-packages.json"
+   * and "nonbrowser-approved-packages.json".  See the Rush documentation for details.
+   */
   "approvedPackagesPolicy": {
+    /**
+     * The review categories allow you to say for example "This library is approved for usage
+     * in prototypes, but not in production code."
+     *
+     * Using the "reviewCategory" field in the "projects" section below, each project can
+     * associated with a review category.  The approval is recorded in the files
+     * "browser-approved-packages.json" and "nonbrowser-approved-packages.json".
+     */
     "reviewCategories": [ "main", "tools", "prototypes" ],
+
+    /**
+     * A list of NPM package scopes that will be excluded from review.
+     * We recommend to exclude TypeScript typings (the "@types" scope), because
+     * if the underlying package was already approved, this would imply that the typings
+     * are also approved.
+     */
     "ignoredNpmScopes": [ "@types" ]
   },
 
+  /**
+   * If you use Git as your version control system, this section has some additional
+   * optional features you can use.
+   */
   "gitPolicy": {
-    // Require GitHub scrubbed e-mails
+    /**
+     * Work at a big company?  Tired of finding Git commits at work with unprofessional Git
+     * emails such as "beer-lover@my-college.edu"?  Rush can validate people's Git email address
+     * before they get started.
+     *
+     * Define a list of regular expressions describing allowable e-mail patterns for Git commits.
+     * They are case-insensitive anchored JavaScript RegExps.  Example: ".*@example\.com"
+     */
     "allowedEmailRegExps": [
       "[^@]+@users\\.noreply\\.github\\.com",
       "travis@example\\.org"
     ],
+
+    /**
+     * When Rush reports that the address is malformed, the notice can include an example
+     * of a recommended email.  Make sure it conforms to one of the allowedEmailRegExps
+     * expressions.
```

Comment:
we might as well do this for them

Reply:
This is technically impossible.  For example, what if the RegExp was `.*microsoft.*`?

---
In reply to: [217227474](https://github.com/Microsoft/rush-example/pull/2#discussion_r217227474) [](ancestors = 217227474)